packages/web/hooks/swap/use-estimate-swap-tx-fees.ts

```diff
@@ -96,9 +97,28 @@ export function useEstimateTxFees({
       wallet?.address !== undefined &&
       typeof wallet?.address === "string",
   });
+
+  // runs the request once
+  const runEstimateTxFeesOnce = useCallback(
+    async (messages: EncodeObject[]) => {
+      if (!wallet) throw new Error(`No wallet found for chain ID: ${chainId}`);
+      return estimateSwapTxFeesQueryFn({
+        wallet,
+        accountStore,
+        messages,
+        apiUtils,
+      });
+    },
+    [accountStore, apiUtils, chainId, wallet]
+  );
```

Comment:
Any reason we have this instead of using the mutation below or just calling the function through an import?

Reply:
`runEstimateTxFeesOnce` is used as an asynchronous function instead of a mutation or directly calling the imported function because:
1. It allows us to immediately execute the estimation and await the result within `finalizeSwapTxData`.
2. We don't need to track the mutation state or trigger side effects based on the result in this context.
3. It keeps the code simpler and more straightforward, as we can directly call and handle the result without additional complexity.
4. We don't require caching or automatic refetching of the result, as the estimation is performed once and used immediately within the function.


Using an asynchronous function here aligns with the specific requirements and usage within finalizeSwapTxData.